### PR TITLE
[xmpp] Improve reconnection logic

### PR DESCRIPTION
**Pull request by lsiepel**
- Improves reconnection logic
 - Improve `Thing` status handling
 - Add configuration parameter `securityMode` this enables the user to have TLS required (as default), optional or disabled.
 - Add null annotations

Fixes: https://github.com/openhab/openhab-addons/issues/9378
Fixes: https://github.com/openhab/openhab-addons/issues/10482 
Fixes: https://github.com/openhab/openhab-addons/issues/14713

Test jar files:
openHAB 4.2.x: https://1drv.ms/u/s!AnMcxmvEeupwjq5fcNrBW_F-fxnlTA?e=R20Ymo

### CODE REVIEW

**Comment by lsiepel** [comment]
@lolodomo are you able to review?

**Comment by lolodomo** [comment]
Will do it during next week.

File: bundles/org.openhab.binding.xmppclient/src/main/java/org/openhab/binding/xmppclient/internal/client/XMPPClientEventlistener.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.xmppclient.internal.client;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link XMPPClientEventlistener} is lib for handling XMPP connection events
```
Comment (lolodomo) [inline bundles/org.openhab.binding.xmppclient/src/main/java/org/openhab/binding/xmppclient/internal/client/XMPPClientEventlistener.java:18]: Rather "interface" ?

File: bundles/org.openhab.binding.xmppclient/README.md [hunk 2]
```diff
@@ -26,22 +26,23 @@ Bridge xmppclient:xmppBridge:xmpp "XMPP Client" [ host="xmpp.example.com", port=
...
+| Name     | Label              | Description                                                        | Required | Default value         |
+|----------|--------------------|--------------------------------------------------------------------|----------|-----------------------|
+| username | Username           | The XMPP username (left part of JID)                               | true     | -                     |
+| domain   | Domain             | The XMPP domain name (right part of JID)                           | true     | -                     |
+| password | Password           | The XMPP user password                                             | true     | -                     |
+| host     | Server Hostname/IP | The IP/Hostname of the XMPP server                                 | false    | as "domain" parameter |
+| port     | XMPP server Port   | The typical port is 5222                                           | false    | 5222                  |
+| port     | Security Mode      | Sets the TLS security mode: `required`, `ifpossible` or `disabled` | false    | `required`            |
```
Comment (lolodomo) [inline bundles/org.openhab.binding.xmppclient/README.md:36]: Typo: parameter name is: securityMode

**Review by lolodomo — CHANGES_REQUESTED** [review]
Just 2 minor comments

File: bundles/org.openhab.binding.xmppclient/src/main/java/org/openhab/binding/xmppclient/internal/client/XMPPClientEventlistener.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.xmppclient/src/main/java/org/openhab/binding/xmppclient/internal/client/XMPPClientEventlistener.java:18]: Fixed.

File: bundles/org.openhab.binding.xmppclient/README.md (hunk 2 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.xmppclient/README.md:36]: Fixed.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**